powerdns.go

```diff
@@ -113,24 +115,29 @@ func New(baseURL string, vHost string, options ...NewOption) *Client {
 }
 
 func parseBaseURL(baseURL string) (string, string, string, error) {
-	u, err := url.Parse(baseURL)
+	parsedURL, err := url.Parse(baseURL)
 	if err != nil {
 		return "", "", "", err
 	}
-	hp := strings.Split(u.Host, ":")
-	hostname := hp[0]
-	var port string
-	if len(hp) > 1 {
-		port = hp[1]
-	} else {
-		if u.Scheme == "https" {
-			port = "443"
-		} else {
-			port = "80"
+
+	hostname, port, err := net.SplitHostPort(parsedURL.Host)
```

Comment:
It's probably too late, but I think we could also use `u.Hostname()` and `u.Port()` instead of `u.Host`:

```go
func parseBaseURL(baseURL string) (string, string, string, error) {
	u, err := url.Parse(baseURL)
	if err != nil {
		return "", "", "", err
	}
	hostname := u.Hostname()
	port := u.Port()
	if port == "" {
		if u.Scheme == "https" {
			port = "443"
		} else {
			port = "80"
		}
	}

	return u.Scheme, hostname, port, nil
}
```

It's working fine for me.

Reply:
This won't catch some invalid entries, e.g. when `baseURL` is `http://::8080`.